src/xscen/ensembles.py

```diff
@@ -682,41 +828,32 @@ def build_partition_data(
     # initialize dict
     subset_kw = subset_kw or {}
     regrid_kw = regrid_kw or {}
+    to_dataset_kw = to_dataset_kw or {}
 
-    list_ds = []
-    for ds in datasets:
-        if subset_kw:
-            ds = subset(ds, **subset_kw)
-
-        if regrid_kw:
-            ds = regrid_dataset(ds, **regrid_kw)
-
-        if indicators_kw:
-            dict_ind = compute_indicators(ds, **indicators_kw)
-            if len(dict_ind) > 1:
-                raise ValueError(
-                    f"The indicators computation should return only indicators of the same frequency.Returned frequencies: {dict_ind.keys()}"
-                )
-            else:
-                ds = list(dict_ind.values())[0]
+    if isinstance(datasets, list):
```

Comment:
Either this should accept both a list or a dict, or the `if` entry for dicts is missing.

Reply:
regarde plus haut, je change les dict en list